src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -304,92 +324,249 @@ class CVulkanLogicalDevice final : public ILogicalDevice
             
     core::smart_refctd_ptr<IGPURenderpass> createGPURenderpass(const IGPURenderpass::SCreationParams& params) override
     {
-        // Todo(achal): Hoist creation out of constructor
-        return nullptr; // return core::make_smart_refctd_ptr<CVulkanRenderpass>(this, params);
+        // auto* vk = m_vkdev->getFunctionTable();
+
+        VkRenderPassCreateInfo createInfo = { VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO };
+        createInfo.pNext = nullptr;
+        createInfo.flags = static_cast<VkRenderPassCreateFlags>(0u); // No flags are supported
+        createInfo.attachmentCount = params.attachmentCount;
+
+        core::vector<VkAttachmentDescription> attachments(createInfo.attachmentCount); // TODO reduce number of allocations/get rid of vectors
```

Comment:
this usage of a vector is actually fine, its out of the hot path and its sized up front.

Reply:
Can I remove the TODO?